File: AndroidWebBox/coconut-network/src/main/kotlin/com/sniper/coconut/network/interceptors/RequestInterceptor.kt
package com.sniper.coconut.network.interceptors

import com.sniper.coconut.network.HttpRequest
import com.sniper.coconut.network.HttpResponse

/**
 * 请求拦截器接口
 * 请求阶段正序执行，响应阶段逆序执行
 */
interface RequestInterceptor {
    /** 请求拦截 */
    suspend fun onRequest(request: HttpRequest): HttpRequest

    /** 响应拦截 */
    suspend fun onResponse(response: HttpResponse): HttpResponse
}

File: iOSWebBox/CoconutNetwork/Sources/CoconutNetwork/interceptors/RequestInterceptor.swift
import Foundation

/// 请求拦截器接口 — 请求阶段正序执行，响应阶段逆序执行。
public protocol RequestInterceptor: Sendable {
    /// 请求拦截（值语义：返回修改后的副本）
    func onRequest(_ request: HttpRequest) async -> HttpRequest
    /// 响应拦截
    func onResponse(_ response: HttpResponse) async -> HttpResponse
}
